e2e/libs/volume/crd.py

```diff
@@ -65,11 +66,20 @@ def create(self, volume_name, size, numberOfReplicas, frontend, migratable, data
                 plural="volumes",
                 body=body
             )
-            if not Standby:
-                self.wait_for_volume_state(volume_name, "detached")
-            else:
+            if Standby:
                 self.wait_for_volume_state(volume_name, "attached")
-            self.wait_for_restore_required_status(volume_name, Standby)
+            else:
+                if fromBackup:
+                    if wait_attached:
+                        self.wait_for_volume_state(volume_name, "attached")
+                    else:
+                        self.wait_for_volume_state(volume_name, "detached")
```

Comment:
If `fromBackup` is not empty and `wait_attached` is `False`,
from https://github.com/longhorn/longhorn-tests/pull/2371/files#r1998210595, does it mean it will implicitly wait for volume restoration complete and the following `And Wait for volume 1 restoration from backup 0 of volume 0 start` will fail?

Reply:
Yes, because when executing `create volume from the backup`, the operation will wait until the volume is created and its state becomes `detached`. Therefore, during the step `Wait for volume 1 restoration from backup 0 of volume 0 start`, it will failed to obtain the restoration state of `volume 1`

Reply:
So why do we need the logic for `fromBackup is not empty and wait_attached is False`?

Reply:
After discussing with @yangchiu we concluded that when creating a volume from  a backup, subsequent keywords must include a check for the `volume wait for detach` action.  For example:https://github.com/longhorn/longhorn-tests/blob/master/e2e/tests/negative/pull_backup_from_another_longhorn.robot#L65-L66